internal/server/data/destination.go

```diff
@@ -176,11 +176,28 @@ func ListDestinations(tx ReadTxn, opts ListDestinationsOptions) ([]models.Destin
 }
 
 func DeleteDestination(tx WriteTxn, id uid.ID) error {
+	dest, err := GetDestination(tx, GetDestinationOptions{ByID: id})
+	if err != nil {
+		return handleError(err)
+	}
+
+	query := querybuilder.New("UPDATE grants")
+	query.B("SET deleted_at = ?,", time.Now())
+	query.B("update_index = nextval('seq_update_index')")
+	query.B("WHERE organization_id = ? AND", tx.OrganizationID())
+	query.B("deleted_at is null")
+	grantsByDestination(query, dest.Name)
```

Comment:
I guess it might make sense to add a `ByDestination` field to `DeleteGrantsOptions`, and use `DeleteGrants` for this. The query looks the same, we're only missing the `ByDestination` option.

Reply:
This is a great suggestion. I went ahead and refactored and added another test.